src/Compilers/CSharp/Portable/Symbols/TypeSymbolWithAnnotations.cs

```diff
@@ -358,8 +347,7 @@ internal TypeSymbolWithAnnotations ToType()
 
         private TypeSymbolWithAnnotations(TypeSymbol defaultType, NullableAnnotation nullableAnnotation, Extensions extensions)
         {
-            Debug.Assert((object)defaultType != null);
-            Debug.Assert(!defaultType.IsNullableType() || (nullableAnnotation != NullableAnnotation.Unknown && nullableAnnotation != NullableAnnotation.NotAnnotated));
+            Debug.Assert(defaultType?.IsNullableType() != true || (nullableAnnotation != NullableAnnotation.Unknown && nullableAnnotation != NullableAnnotation.NotAnnotated));
```

Comment:
>Debug.Assert(defaultType [](start = 12, length = 24)

We're opening to door to having an annotation when the `defaultType` is null. Are there any restrictions in that case? If there are some, it'd be great to assert them.

Reply:
No, there are no restrictions.  In flow analysis, an expression such as `null` has no type but may have an annotation (of Nullable) for scenarios in which it is erroneously used as an lvalue.

---
In reply to: [260034302](https://github.com/dotnet/roslyn/pull/33648#discussion_r260034302) [](ancestors = 260034302)